### DISCUSSION TITLE
double nested arrays

### DISCUSSION THREAD

**Discussion by luckydonald** [spatie/laravel-data#476] [Q&A]
I have to support the following json array:

```php 
$rules = [
  'items' => 'array',
  'items.*' => 'array',
  'items.*.*.name' => 'string',
  // more stuff an item has
];
```

For example it would look like that in Json:
```json
{
"items": {
   "salty": {
     "meat": {"name": "Beef Jerky"},
     "plant": {"name": "Sea Weed"},
   },
  "sweet": {
     "fruit": {"name": "Banana"},
     "dessert": {"name": "Mousse au Chocolat"}
   }
}
```

I made a class `Item` for that `{"name": "…"}` part, and a class `Outer` for the overall object.

Now I need to define `Outer` and tell it that I want to have `Item[][]` as the `$items` variable:

I tried to use `DataCollectionOf` twice, but that doesn't seem to work:

```php
<?php

namespace App\Data;

use Spatie\LaravelData\Attributes\DataCollectionOf;

readonly class Outer
{
    public function __construct(
        #[DataCollectionOf(are DataCollectionOf(Item::class))]
        public array $items,  // How do I tell this I want `Item[][]`, both having string keys
    )
    {
    }
}
```

This results in the following error:
```
Expected parameter of type 'string', '\Spatie\LaravelData\Attributes\DataCollectionOf' provided 
```

**Comment by rubenvanassche** [comment]
Hmm I don't see how we can support this. Every layer needs a data object so you could have Item and ParentItem? Other solution could be to implement a custom cast.
